pytest/lib/peer.py

```diff
@@ -179,6 +179,13 @@ async def run_handshake(conn: Connection,
         await conn.send(handshake)
         response = await conn.recv()
```

Comment:
Since we are only expecting `Handshake` or `HandshakeFailure` we can specify `expected` field in `conn.recv`

```suggestion
        response = await conn.recv(lambda msg : msg.enum.startswith('Handshake')
```

Reply:
This needs to be done in the **three** calls to `conn.recv` in this function.

Reply:
> This needs to be done in the three calls to conn.recv in this function.

In practice the first two will always be HandshakeFailure but sure.